### ISSUE TITLE
Support for C++

### ISSUE THREAD

**Issue by FramusRock** [citrusbyte/Auto-Importer-for-Xcode#26]
Hey,

I absolutely love your tool and use it everyday!

Is it somehow possible (maybe with a setting or auto-recognize) to use #include instead of #import when coding C++ in Xcode?

**Comment by lucholaf** [comment]
@FramusRock good idea thanks, are you using .cpp and/or .mm files for c++? btw, you're welcome to submit a pull request if you want to implement it.

**Comment by FramusRock** [comment]
In .mm files #import works just fine. I'm talking about .cpp in which #include is a must. Auto-Importer works just fine there, it should just write #include instead of #import ;)

**Comment by FramusRock** [comment]
I looked into it a little, maybe you can help me out, so that I can make an appropriate pull request.

In LAFIDESourceCodeEditor.m, there is this:

- (NSString *)importStatementFor:(NSString *)header {
    return [NSString stringWithFormat:@"#import \"%@\"", header];
}

How can I find out if the current file that this header belongs to is a CPP file or ending in .cpp?

**Comment by FramusRock** [comment]
I'm still interested in it, could you give me a hint how I can achieve it, please?

**Comment by lucholaf** [comment]
hey @FramusRock I spent some time trying to figure out how to get the filename from current editor but still no luck. Another approach would be to get the first existing import/include line on the file and use that to decide if you should be using #import or #include

**Comment by FramusRock** [comment]
hey @lucholaf , thank you. I also spent some time on it, but also didn't have any luck getting it right.
What you're saying sounds like a good idea. I'll have a try and report back.

**Comment by lucholaf** [comment]
@FramusRock +1

**Comment by FramusRock** [comment]
I made it! Pull request sent!
